src/js/models/layerstructure.js

```diff
@@ -203,6 +203,14 @@ define(function (require, exports, module) {
                     return layer.kind === layer.layerKinds.GROUPEND;
                 });
         },
+        
+        /**
+        * All non-endgroup layers, in reverse order
+        * @type {Immutable.List.<Layer>}
+        */
+        "allVisibleReversed": function () {
+            return this.allVisible.reverse();
```

Comment:
Any idea if `.reverse` is `O(1)` or `O(n)`? If it's the former, it might be better to just call `.reverse()` at the point of use. If it's the latter and it's actually expensive then instead it might be better to make this a derived (cached) property.